### DISCUSSION TITLE
Need help with Quasar + Electron + RxDB

### DISCUSSION THREAD

**Discussion by krzjablonski** [quasarframework/quasar#10796] [CLI - Electron mode]
Hi! 

I need an offline database solution for electron and android. Therefore I chose to try RxDB, as due to many adapters it can be used on both platforms. Unfortunately on Electron I came across an error that kills the site. 

Error is produced when calling addCollections on RxDB instance. The screen goes white and devtools are disconnected. 

Here is a link for a public github repo with a sample project that has the same issue: https://github.com/krzjablonski/quasar_rxdb
I wonder if that is because I expose database object through preload script?

I would appreciate any help 😊

A couple of additional info:
Node version: 14.15.0
Electron version: 15.2.0
Os: Windows 10 Pro 21H1 19043.1237

Thanks!

**Comment by ontwikkelfabriek** [comment]
@krzjablonski 

I ran your code in electron mode and have the same issue. 

I don't know what the problem is but I do know this Quasar V1 repo/app ( pretty awesome app ) uses Electron/Capacitor/RxDB and works: 
https://github.com/vycoder/qodo

**Reply by krzjablonski** [reply]
I've checked this project and for electron, it's using `pouchdb-adapter-idb` adapter which has nothing to do with NodeJs. Non the less will try it out when I find some time for it. Thanks :-)

**Reply by krzjablonski** [reply]
Tried this adapter and it works, but tbh if I would like to use IndexedDB I would use something like Dexie.js or another IndexedDB wrapper.

**Comment by pikil** [comment]
@krzjablonski 

I haven't run the app yet, but I do have a question first:
What was the reason to run rxdb through electron-preload.js file? Can you just import it directly and see if there is any difference?

Just curious what would the outcome be...

**Reply by krzjablonski** [reply]
Hi @pikil. It's exposed by electron-preload so I do not have to expose the whole nodejs module system to the renderer process. I wanted to follow the `contextInsolation: true` and `nodeIntergration: false` approach. On the other hand exposing whole database system is also not the most secure option whatsoever ;-) 

I will check if allowing the renderer process to access the node will help with that and certainly give you an update. 

Thanks!

**Reply by pikil** [reply]
Am I looking at the wrong thing? I was under the impression that RxDB is not a node script initially (it does the node plugin though: https://rxdb.info/tutorials/server.html)...

Have a look at the example below, it is not having a node-js implementation:
https://github.com/pubkey/rxdb/tree/master/examples/vanilla

Anyway, you can try the node plugin to see if that helps :)

**Reply by krzjablonski** [reply]
The problem is that when using [leveldown adapter](https://rxdb.info/adapters.html#leveldown) it fires `require()` on node modules which is not allowed in the renderer process unless I enable node integration :-)
